src/connmgr/connection.rs

```diff
@@ -2012,14 +2015,11 @@ where
                 let max_read = cmp::min(tmp_buf.len(), zsess_in.credits() as usize);
 
                 let (size, done) = match req_body.try_recv(&mut tmp_buf[..max_read])? {
-                    RecvStatus::Complete((), written) => (written, true),
-                    RecvStatus::Read((), written) => {
-                        if written == 0 {
-                            add_to_buffer.set(Some(req_body.add_to_buffer()));
-                            continue;
-                        }
-
-                        (written, false)
+                    RecvStatus::Complete((), size) => (size, true),
+                    RecvStatus::Read((), size) => (size, false),
+                    RecvStatus::NeedBytes(()) => {
+                        add_to_buffer.set(Some(req_body.add_to_buffer()));
+                        continue;
                     }
                 };
```

Comment:
This code reads request body bytes from a client into a temporary buffer that is guaranteed to be non-zero-sized. Previously, it would try to read more bytes into the input buffer whenever `try_recv` returned zero, which could only happen whenever bytes were needed since the output buffer would always have room. Now, it only tries to read more bytes into the input buffer whenever `try_recv` returns `NeedBytes`. This is identical behavior as before so there should be no performance impact.